EIPS/eip-erc_5219_resolve_mode.md

```diff
@@ -0,0 +1,58 @@
+---
+title: ERC-5219 Resolve Mode
+description: Adds an ERC-4804 resolveMode to support ERC-5219 requests
+author: Gavin John (@Pandapip1), Qi Zhou (@qizhou)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-04-27
+requires: 4804, 5219
+---
+
+## Abstract
+
+This EIP adds a new [ERC-4804](./eip-4804.md) `resolveMode` to resolve [ERC-5219](./eip-5219.md) contract resource requests.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Contracts wishing to use ERC-5219 as their ERC-4804 resolve mode must implement the following interface:
+
+```solidity
+/// @dev IDecentralizedApp is the ERC-5219 interface
+interface IERC5219Resolver is IDecentralizedApp {
+    // @notice The ERC-4804 resolve mode
+    // @dev    This MUST return "erc5219" for ERC-5219 resolution (case-insensitive). The other options, as of writing this, are "auto" for automatic resolution, or "manual" for manual resolution.
+    function resolveMode() external pure returns (string mode);
```

Comment:
It seems to a bit weird, but the original return type of `resolveMode()` is `bytes32` rather than `string` (as clarified in the amendment PR https://github.com/ethereum/EIPs/pull/6860/files), and returning a string in solidity will automatically convert it to `bytes32` (e.g., `return "auto"`).

Since some applications/contracts are deployed using `bytes32` for `resolveMode`, I think we should stick to the conventional for compatibility.  But in the future, we could amend ERC-4804 to support both two types (string, byte32) for the return type of `resolveMode()` if len(output) == 32, then parse it as bytes32-string, otherwise parse is as string directly.